pkg/registry/common/nslabelsmatch/client_test.go

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) 2021 Doc.ai and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package nslabelsmatch_test
+
+import (
+	"context"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"go.uber.org/goleak"
+
+	"github.com/networkservicemesh/api/pkg/api/registry"
+
+	"github.com/networkservicemesh/sdk/pkg/registry/common/memory"
+	"github.com/networkservicemesh/sdk/pkg/registry/common/nslabelsmatch"
+	"github.com/networkservicemesh/sdk/pkg/registry/core/adapters"
+	"github.com/networkservicemesh/sdk/pkg/registry/core/chain"
+)
+
+const (
+	testEnvName                      = "NODE_NAME"
+	testEnvValue                     = "testValue"
+	destinationTestKey               = "nodeName"
+	destinationTestCorrectTemplate   = "{{index .Src \"NodeNameKey\"}}"
+	destinationTestIncorrectTemplate = "{{index .Src \"SomeRandomKey\"}}"
+	destinationTestNotATemplate      = "SomeRandomString"
+)
+
+func TestNSTemplates(t *testing.T) {
+	t.Cleanup(func() { goleak.VerifyNone(t) })
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*5)
+	defer cancel()
+
+	err := os.Setenv(testEnvName, testEnvValue)
+	require.NoError(t, err)
+
+	nsrc := chain.NewNetworkServiceRegistryClient(
+		nslabelsmatch.NewNetworkServiceRegistryClient(ctx),
+		adapters.NetworkServiceServerToClient(memory.NewNetworkServiceRegistryServer()),
+	)
+
+	type test struct {
+		name     string
+		template string
+		want     string
+	}
+
+	tests := []test{
```

Comment:
Please use here slice of anonymous structs:
```go
tests := []struct{
    name string
    template string
    want string
}{
    {
        name: "CorrectTemplate",
        ...
    }
}
```

Reply:
I think it better to use static arrays here..
```
tests := [...]struct{
    name string
    template string
    want string
}{
    {
        name: "CorrectTemplate",
        ...
    }
}
```